src/practices/Go/GoGitignoreCorrectlySetPractice.ts

```diff
@@ -0,0 +1,101 @@
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import {
+  PracticeEvaluationResult,
+  PracticeImpact,
+  ProgrammingLanguage,
+} from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { PracticeBase } from '../PracticeBase';
+import { ReportDetailType } from '../../reporters/ReporterData';
+import { FixerContext } from '../../contexts/fixer/FixerContext';
+
+@DxPractice({
+  id: 'Go.GitignoreCorrectlySet',
+  name: 'Set .gitignore Correctly',
+  impact: PracticeImpact.high,
+  suggestion: 'Set patterns in the .gitignore as usual.',
+  reportOnlyOnce: true,
+  url: 'https://github.com/github/gitignore/blob/master/Node.gitignore',
+  dependsOn: { practicing: ['LanguageIndependent.GitignoreIsPresent'] },
+})
+export class GoGitignoreCorrectlySetPractice extends PracticeBase {
+  private parsedGitignore: string[] = [];
+
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.Go;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (!ctx.root.fileInspector) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const parseGitignore = (gitignoreFile: string) => {
+      return gitignoreFile
+        .toString()
+        .split(/\r?\n/)
+        .filter((content) => content.trim() !== '' && !content.startsWith('#'));
+    };
+    const content = await ctx.root.fileInspector.readFile('.gitignore');
+    const parsedGitignore = parseGitignore(content);
+    this.parsedGitignore = parsedGitignore;
+
+    // node_modules
+    const binaryFilesRegex = parsedGitignore.find((value: string) =>
+      /.exe$|.exe~$|.dll$|.so$|.dylib$|.test$/.test(value),
```

Comment:
See the comment here https://github.com/DXHeroes/dx-scanner/issues/407#issuecomment-714479986

Reply:
Changed this in the new PR made